src/odemis/driver/test/technolution_test.py

```diff
@@ -0,0 +1,1460 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on 11 May 2020
+
+@author: Sabrina Rossberger, Kornee Kleijwegt
+
+Copyright © 2019-2020 Kornee Kleijwegt, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+
+These test cases can only be done using the simulator for the ASM.
+After installing the simulator it can be starting using the following commands in the terminal:
+    sudo su
+    echo 134217728 > /proc/sys/net/core/rmem_max;
+    systemctl restart vsftpd.service; systemctl restart asm_service; systemctl restart sam_simulator;
+    systemctl status asm_service;
+"""
+import math
+import os
+import time
+import logging
+import unittest
+import numpy
+import matplotlib.pyplot as plt
+
+from odemis import model
+from odemis.util import almost_equal
+
+from openapi_server.models import CalibrationLoopParameters
+from openapi_server.models.mega_field_meta_data import MegaFieldMetaData
+
+from odemis.driver.technolution import AcquisitionServer, convertBits, convertRange, DATA_CONTENT_TO_ASM, INT16
+
+# Export TEST_NOHW = 1 to prevent using the real hardware
+TEST_NOHW = (os.environ.get("TEST_NOHW", "0") != "0")  # Default to Hw testing
+
+# How often tests holding randomized input are repeated (to catch all rounding, floating point errors etc.)
+REPEAT_1000 = range(0, 1000)
```

Comment:
For test defining a global variable is maybe a bit too much as it is never used except here in these tests? Might be enough to just define the range in the respective test cases?